pkg/kubelet/kubelet.go

```diff
@@ -1522,6 +1522,9 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType, pod, mirrorPod *v1.Pod, podStatus *kubecontainer.PodStatus) (isTerminal bool, err error) {
 	klog.V(4).InfoS("syncPod enter", "pod", klog.KObj(pod), "podUID", pod.UID)
 	defer func() {
+		if _, exists := kl.podManager.GetPodByUID(pod.UID); !exists {
```

Comment:
Just at a first glance, it is wrong to check state here, we should fix the reconciliation back where the bug happened (kubelet config -> pod worker).  It’s the responsibility of that layer to correctly send us REMOVE events correctly, when then would propagate to here.